rosdoc2/verbs/build/builders/sphinx_builder.py

```diff
@@ -48,30 +71,51 @@ def ensure_global(name, default):
 if rosdoc2_settings.get('enable_autodoc', True):
     print('[rosdoc2] enabling autodoc', file=sys.stderr)
     extensions.append('sphinx.ext.autodoc')
+    # Provide all runtime dependencies to be mocked up
+    # Note: `autodoc` only mocks up those modules that it actually cannot locate in PATH
+    autodoc_mock_imports = {exec_depends}
+    # Add the package directory to PATH, so that `sphinx-autodoc` can import it
+    sys.path.insert(0, os.path.dirname('{package_src_directory}'))
 
 if rosdoc2_settings.get('enable_intersphinx', True):
     print('[rosdoc2] enabling intersphinx', file=sys.stderr)
     extensions.append('sphinx.ext.intersphinx')
 
-if rosdoc2_settings.get('enable_breathe', True):
+build_type = '{build_type}'
+always_run_doxygen = {always_run_doxygen}
+# By default, the `exhale`/`breathe` extensions should be added if `doxygen` was invoked
+is_doxygen_invoked = build_type in ('ament_cmake', 'cmake') or always_run_doxygen
+
+if rosdoc2_settings.get('enable_breathe', is_doxygen_invoked):
     # Configure Breathe.
     # Breathe ingests the XML output from Doxygen and makes it accessible from Sphinx.
     print('[rosdoc2] enabling breathe', file=sys.stderr)
+    # First check that doxygen would have been run
+    if not is_doxygen_invoked:
+        raise RuntimeError(
+            "Cannot enable the 'breathe' extension if 'doxygen' is not invoked."
+            "Please enable 'always_run_doxygen' if the package is not an"
+            "'ament_cmake' or 'cmake' package.")
     ensure_global('breathe_projects', {{}})
-    breathe_projects.update({{
-{breathe_projects}}})
+    breathe_projects.update({{{breathe_projects}}})
     if breathe_projects:
         # Enable Breathe and arbitrarily select the first project.
         extensions.append('breathe')
         breathe_default_project = next(iter(breathe_projects.keys()))
 
-if rosdoc2_settings.get('enable_exhale', True):
+if rosdoc2_settings.get('enable_exhale', is_doxygen_invoked):
     # Configure Exhale.
     # Exhale uses the output of Doxygen and Breathe to create easier to browse pages
     # for classes and functions documented with Doxygen.
     # This is similar to the class hierarchies and namespace listing provided by
     # Doxygen out of the box.
     print('[rosdoc2] enabling exhale', file=sys.stderr)
+    # First check that doxygen would have been run
+    if not is_doxygen_invoked:
+        raise RuntimeError(
+            "Cannot enable the 'breathe' extension if 'doxygen' is not invoked."
+            "Please enable 'always_run_doxygen' if the package is not an"
```

Comment:
```suggestion
            "Cannot enable the 'breathe' extension if 'doxygen' is not invoked. "
            "Please enable 'always_run_doxygen' if the package is not an "
```

Reply:
Addressed in https://github.com/ros-infrastructure/rosdoc2/pull/28/commits/fb0eab5d53133b0a0d71a59edb7213a36c6fb522